### ISSUE TITLE
Add TypeScript Support to the Project

### ISSUE THREAD

**Issue by Danguya** [fisshy/react-scroll#565]
Hello team,

I'd like to propose adding TypeScript support to our project.
Currently, we are using older versions of several libraries, such as webpack, babel, karma, and ECMAScript.
I believe introducing TypeScript will help us write safer and more readable code, leveraging the benefits of static typing.

Reasons to adopt TypeScript:

Static Typing: This will help us catch errors at compile-time rather than run-time.
Readability: TypeScript can make our code more readable, especially when dealing with complex data structures.
Self-documenting: With TypeScript, we can have better self-documentation through types.
IDE Integration: Many IDEs have robust support for TypeScript, which can enhance developer productivity.
Proposed Tasks:

- [ ]  Update libraries to the latest versions that are TypeScript compatible (webpack, babel, karma, etc.).
- [ ]  Set up TypeScript in the project.
- [ ]  Refactor existing files to use TypeScript.
- [ ]  Update project documentation to reflect the changes.
- [ ] I'd love to hear feedback from the team on this proposal. If everyone agrees, I'm willing to assist in the migration and setup.

Thank you!

**Comment by leandroruel** [comment]
is this? https://www.npmjs.com/package/@types/react-scroll

**Comment by Danguya** [comment]
@leandroruel  

No, I mean for us to gradually migrate the 'react-scroll' project to TypeScript, as this will make it easier for us to use the latest ECMAScript features, resulting in simpler and more modern JavaScript code. It will also allow us to move away from direct use of Babel and Webpack. I believe this will make the project more up-to-date and appealing for more developers to contribute.
